src/app/item/analyze/page.tsx

```
@@ -97,6 +118,423 @@ export default function Home() {
+    if (itemData.weights && itemData.weights.length > 0) {
+      console.log('[DEBUG] Checking threshold for each weight...');
+      itemData.weights.forEach(weight => {
+        // Only count identifications in the weight where the weight value is >= 0.03 (3%)
+        const count = Object.values(weight.identifications).filter(val => Math.abs(val) >= 0.03).length;
+        validIdsCountPerWeight[weight.weight_id] = count;
+      });
+    }
+    // Fallback: use all weight identifications if no weights (rare)
+    let fallbackValidIdsCount = 0;
+    if (!itemData.weights || itemData.weights.length === 0) {
+      if (itemData.input && itemData.input.identifications) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)